src/CalcViewModel/UnitConverterViewModel.cpp

```diff
@@ -53,6 +53,8 @@ constexpr size_t SELECTED_TARGET_UNIT = 2;
 
 // x millisecond delay before we consider conversion to be final
 constexpr unsigned int CONVERSION_FINALIZED_DELAY_IN_MS = 1000;
+const wregex regexTrimSpacesStart = wregex(L"^\\s+");
```

Comment:
A few thoughts: I think what is considered a whitespace character (the `\s`) by the std::wregex library depends on the current locale setting. So there might be some odd cases where `\u00a0` isn't consider whitespace character by the locale, yet shows up in the currency result.

Reply:
I had a doubt too, but it should be fine. If no flags are passed in the constructor, the wregex will use ECMA and based on the spec, `\s` will always match `\u00a0`.

Spec here:
http://ecma-international.org/ecma-262/5.1/#sec-7.2

Reply:
Nice! Thanks for the reply.

I suppose the problem might be if locales start using Narrow No-break Space (U+202F) at some point.
See the "Migration" section in CLDR v34 here: http://cldr.unicode.org/index/downloads/cldr-34

However, we currently override this in the CLDR data used by NLS in Windows, so that won't be an issue for now...

Reply:
Actually, upon second reading of the ECMAScript spec:

> ECMAScript implementations must recognise all of the white space characters defined in Unicode 3.0. Later editions of the Unicode Standard may define other white space characters. ECMAScript implementations may recognise white space characters from later editions of the Unicode Standard.

U+202F `NARROW NO-BREAK SPACE` was added in Unicode 3.0.0, and is a Space (Zs) category. So I guess this *should* work for U+202F as well.

Reply:
Thanks for this thorough and thoughtful review, @jefgen.  Much appreciated.